src/shared/istiopolicy/creator.go

```diff
@@ -98,6 +102,110 @@ func (c *Creator) Create(
+	serviceAccountLabelValue, ok := clientIntents.Annotations[v1alpha2.OtterizeClientServiceAccountAnnotation]
+	if ok && serviceAccountLabelValue == clientServiceAccount {
+		return nil
+	}
+
+	updatedIntents := clientIntents.DeepCopy()
+	if updatedIntents.Annotations == nil {
+		updatedIntents.Annotations = make(map[string]string)
+	}
+
+	updatedIntents.Annotations[v1alpha2.OtterizeClientServiceAccountAnnotation] = clientServiceAccount
+	err := c.client.Patch(ctx, updatedIntents, client.MergeFrom(clientIntents))
+	if err != nil {
+		logrus.WithError(err).Errorln("Failed updating intent with service account name")
+		c.recorder.RecordWarningEventf(clientIntents, ReasonIntentsLabelFailed, "Failed updating intent with service account name: %s", err.Error())
+		return err
+	}
+
+	logrus.Infof("updating intent %s with service account name %s", clientIntents.Name, clientServiceAccount)
+
+	c.recorder.RecordNormalEventf(clientIntents, ReasonServiceAccountFound, "Client intents service account found %s", clientServiceAccount)
+	return nil
+}
+
+func (c *Creator) updateSharedServiceAccountsInNamespace(ctx context.Context, namespace string) error {
+	var namespacesClientIntents v1alpha2.ClientIntentsList
+	err := c.client.List(
+		ctx, &namespacesClientIntents,
+		&client.ListOptions{Namespace: namespace})
+	if err != nil {
+		return err
+	}
+
+	clientsByServiceAccount := lo.GroupBy(namespacesClientIntents.Items, func(intents v1alpha2.ClientIntents) string {
+		return intents.Annotations[v1alpha2.OtterizeClientServiceAccountAnnotation]
+	})
+
+	for clientServiceAccountName, clientIntents := range clientsByServiceAccount {
+		err = c.updateServiceAccountSharedStatus(ctx, clientIntents, clientServiceAccountName)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Creator) updateServiceAccountSharedStatus(ctx context.Context, clientIntents []v1alpha2.ClientIntents, serviceAccount string) error {
+	logrus.Infof("Found %d intents with service account %s", len(clientIntents), serviceAccount)
+	isServiceAccountShared := len(clientIntents) > 1
+	sharedAccountValue := lo.Ternary(isServiceAccountShared, "true", "false")
+
+	clients := lo.Map(clientIntents, func(intents v1alpha2.ClientIntents, _ int) string {
+		return intents.Spec.Service.Name
+	})
+	clientsNames := strings.Join(clients, ",")
+
+	for _, intents := range clientIntents {
+		if !shouldUpdateStatus(intents, serviceAccount, sharedAccountValue) {
+			continue
+		}
+
+		updatedIntents := intents.DeepCopy()
+
+		updatedIntents.Annotations[v1alpha2.OtterizeSharedServiceAccountAnnotation] = sharedAccountValue
+		err := c.client.Patch(ctx, updatedIntents, client.MergeFrom(&intents))
+		if err != nil {
+			return err
+		}
+
+		if isServiceAccountShared {
+			c.recorder.RecordWarningEventf(updatedIntents, ReasonSharedServiceAccount, "Service account %s is shared and will grant access to the following clients %s", serviceAccount, clientsNames)
```

Comment:
Can you make it: "...and will *also* grant access to the following clients*:*"
(differences: 'also' and ':')